### PR TITLE
#838 K8sClient need add command and args arguments

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #838)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [x] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SClient.scala [hunk 1]
```diff
@@ -212,44 +213,46 @@ object K8SClient {
           private[this] var labelName: String = _
           private[this] var envs: Map[String, String] = Map.empty
           private[this] var ports: Map[Int, Int] = Map.empty
+          private[this] var command: Seq[String] = Seq.empty
+          private[this] var args: Seq[String] = Seq.empty
 
           override def name(name: String): ContainerCreator = {
-            this.name = name
+            this.name = CommonUtils.requireNonEmpty(name)
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SClient.scala]: great!

File: ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SJson.scala [hunk 2]
```diff
@@ -110,20 +110,30 @@ object K8SJson {
                                       image: String,
                                       env: Seq[CreatePodEnv],
                                       ports: Seq[CreatePodPortMapping],
-                                      imagePullPolicy: ImagePullPolicy)
+                                      imagePullPolicy: ImagePullPolicy,
+                                      command: Seq[String],
+                                      args: Seq[String])
 
   implicit val CREATEPOD_CONTAINER_FORMAT: RootJsonFormat[CreatePodContainer] = new RootJsonFormat[CreatePodContainer] {
     override def write(obj: CreatePodContainer): JsValue = JsObject(
```
Comment (chia7712) [inline ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SJson.scala]: Why we implement the json serialization manually? ping @jackyoh.

Comment (jackyoh) [reply ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SJson.scala]: 因為 CreatePodContainer 的 imagePullPolicy 所要求的型態是 ImagePullPolicy, 所以我才會自己 implement, 不確定有沒有更好的作法.

Comment (chia7712) [reply ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SJson.scala]: You should implement json format for only ImagePullPolicy. I will file a issue for it

Comment (jackyoh) [reply ohara-agent/src/main/scala/com/island/ohara/agent/k8s/K8SJson.scala]: OK, Thank you.

**Pull request merged**